### ISSUE TITLE
Resource release in cloned trajectories is not working

### ISSUE THREAD

**Issue by lucicd** [r-simmer/simmer#105]
Example below simulates two dependent resources - res1 and res2. Arrival seizes res1 and releases it after 7 time units but only only after res2 is seized as well. res2 is then, independently of res1, released after 90 time units.

![image](https://user-images.githubusercontent.com/1948168/27277823-9bf0be0e-54f0-11e7-8761-adc05cc8417a.png)

```
trj <- trajectory() %>%
    seize("res1", 1) %>%
    seize("res2", 1) %>%
    clone(2, 
          trajectory() %>%
              timeout(7) %>%
              release("res1"),
          trajectory() %>%
              timeout(90) %>%
              release("res2")) %>%
    synchronize()

env <- simmer(verbose = TRUE) %>%
    add_resource("res1", 2) %>%
    add_resource("res2", 6) %>%
    add_generator("request", trj, at(0)) %>%
    run()
```

Unfortunately, example does not work because res2 can't be released in the cloned trajectory. This is the output of the simulation.

![image](https://user-images.githubusercontent.com/1948168/27278005-3c386ae2-54f1-11e7-9e74-77d0c571076b.png)

I'm using Windows 7, R x64 3.4.0 and simmer 3.6.2.

**Comment by Enchufa2** [comment]
It doesn't work because it shouldn't by design. Perhaps we should state this more clearly in the documentation, but this is an intended behaviour: every resource seized or batch joined prior to the cloning is the the _original_ arrival's sole responsibility, i.e., the one that takes the first trajectory in the `clone()`. Therefore, this works:

```r
trj <- trajectory() %>%
  seize("res1", 1) %>%
  seize("res2", 1) %>%
  clone(2, 
        trajectory() %>%
          timeout(7) %>%
          release("res1") %>%
          release("res2"),
        trajectory() %>%
          timeout(90)) %>%
  synchronize()
```

The reason for this is consistency: clones are arrivals under a common name, but they are separate entities, so they couldn't seize any resource or join any batch before existing. Moreover, if we had decided to allow the clones to interact with the resources held by the _original_, it is not clear how to handle this: you cannot artificially seize more resources for the clones, and you shouldn't allow them to compete for the `release()`. 

In a nutshell, we decided to create the clones free from previous seizes and batches. If you need the clones to take responsibility for different resources, then you should rewrite your trajectory as follows:

```r
trj <- trajectory() %>%
  clone(2, 
        trajectory() %>%
          seize("res1", 1) %>%
          timeout(7) %>%
          release("res1"),
        trajectory() %>%
          seize("res2", 1) %>%
          timeout(90) %>%
          release("res2")) %>%
  synchronize()
```

**Comment by lucicd** [comment]
Thank you for your kind explanation. I now know why it doesn't work and will think of some workaround.
